internal/cmdliveinit/cmdliveinit_test.go

```diff
@@ -223,17 +290,39 @@ func TestCmd_Run(t *testing.T) {
 				return
 			}
 
-			// Otherwise, validate the kptfile values
+			// Otherwise, validate the kptfile values and/or resourcegroup values.
+			var actualInv kptfilev1.Inventory
 			assert.NoError(t, err)
 			kf, err := pkg.ReadKptfile(w.WorkspaceDirectory)
 			assert.NoError(t, err)
-			if !assert.NotNil(t, kf.Inventory) {
-				t.FailNow()
+
+			switch tc.rgfilename {
```

Comment:
if/else would be much better from readability perspective

Reply:
I'd argue that it is more idiomatic to write Go code using switch cases rather than if/else for branching logic ([https://go.dev/doc/effective_go#switch](https://go.dev/doc/effective_go#switch)) especially since there is also nested if logic branches.
Also, this logic branch will be flattened once we expose the UX and surface changes to users since we will always need the resourcegroup file checking. I don't think we need to worry too much about this given that this is transient code until we launch.
While I favor switch cases, I am happy to convert it to if/else if there is strong preference.

Reply:
Just to explain the readability rationale a little bit. `switch` is better serving the branching logic, totally agree. The `swich` will work if "RG" itself has the branching logic. But I think here the Rg and inventory are two factors that need to work together to decide the branches, right? I expect the `if/else` code can be read as "if RG is set, use RG; otherwise try using inventory; if neither works, fail". 
 
Since this is unittest, the code readability is not as critical as functioning code. I'm fine with either way.
 defer to you . :)

Reply:
Thanks for the explanation - that makes sense! Since a follow-up PR will remove this logic branch soon anyway, I'll just leave it for now.